src/Storages/MergeTree/MergeTreeIndexVectorSimilarity.cpp

```diff
@@ -347,53 +347,58 @@ void MergeTreeIndexAggregatorVectorSimilarity::update(const Block & block, size_
     if (index_sample_block.columns() > 1)
         throw Exception(ErrorCodes::LOGICAL_ERROR, "Expected block with single column");
 
-    const String & index_column_name = index_sample_block.getByPosition(0).name;
```

Comment:
Problem was that l. 350 and l. 385 assumed that the first column in the insert-data-block contains a vector similarity index. Made the logic consistent with the other skipping indexes which resolves the issue.